akka-http/client/src/main/scala/endpoints4s/akkahttp/client/ChunkedEntities.scala

```diff
@@ -65,26 +70,59 @@ trait ChunkedJsonEntities
 
   def jsonChunksRequest[A](implicit
       codec: JsonCodec[A]
+  ): RequestEntity[Chunks[A]] = jsonChunksRequest(noopFraming)
+
+  def jsonChunksRequest[A](framing: Framing)(implicit
+      codec: JsonCodec[A]
   ): RequestEntity[Chunks[A]] = {
     val encoder = stringCodec(codec)
     chunkedRequestEntity(
       ContentTypes.`application/json`,
-      a => ByteString.fromString(encoder.encode(a))
+      a => ByteString.fromString(encoder.encode(a)),
+      framing.request
     )
   }
 
   def jsonChunksResponse[A](implicit
       codec: JsonCodec[A]
+  ): ResponseEntity[Chunks[A]] = jsonChunksResponse(noopFraming)
+
+  def jsonChunksResponse[A](framing: Framing)(implicit
+      codec: JsonCodec[A]
   ): ResponseEntity[Chunks[A]] = {
     val decoder = stringCodec(codec)
-    chunkedResponseEntity { byteString =>
-      val string = byteString.utf8String
-      decoder
-        .decode(string)
-        .toEither
-        .left
-        .map(errors => new Throwable(errors.mkString(". ")))
-    }
+    chunkedResponseEntity(
+      { byteString =>
+        val string = byteString.utf8String
+        decoder
+          .decode(string)
+          .toEither
+          .left
+          .map(errors => new Throwable(errors.mkString(". ")))
+      },
+      framing.response
+    )
+  }
+
+  trait Framing {
+    def request: Flow[ByteString, ByteString, NotUsed]
+    def response: Flow[ByteString, ByteString, NotUsed]
```

Comment:
It would be great to document what is expected from these flows. Especially since their input and output types are both `ByteString`, it is easy to implement them with the “identity” flow, but that would lead to incorrect results. I think it is important to say that the request flow takes as input the frames, and outputs the frames “wrapped” in a specific way.

Reply:
Would it be best to document this on algebra or on each interpreter?

Reply:
https://github.com/endpoints4s/endpoints4s/pull/972#discussion_r773998549 comment is about the same thing, as I understand.

Reply:
On the algebra, it is good to mention how client, server, and documentation interpreters must interpret the member. Like here, for instance:

https://github.com/endpoints4s/endpoints4s/blob/9493c2edf1e4d488228d98805104075e106bb2b0/algebras/algebra/src/main/scala/endpoints4s/algebra/Requests.scala#L132-L141

On interpreters, it is good to underline the important details that matter for the users of that interpreter, in case they want to customize, integrate, compose, etc.

Reply:
I added additional documentation on the algebra level, but I am not too sure about what to add on interpreter level that is not already said on the algebra level. I was trying to not unnecessarily overexplain it, since I assume that developers implementing their own framing algorithms will have to have some prior knowledge for what framing is and will need to write their own tests too. But I am definitely open to any suggestions for documentation improvements!

One small thing to have in mind is that framing can do more than just wrap them, for example it could also chop them up when encoding for more efficient transport.

Reply:
I think what you did is great!